theano/misc/pkl_utils.py

```diff
@@ -37,7 +37,11 @@
 __copyright__ = "Copyright 2013, Universite de Montreal"
 __license__ = "3-clause BSD"
 
-sys.setrecursionlimit(3000)
+
+min_recursion = 3000
+if sys.getrecursionlimit() < min_recursion:
+    sys.setrecursionlimit(min_recursion)
+
```

Comment:
I just figure if the user wants to set a higher recursion limit on their system Theano shouldn't prevent them. Many systems can support much larger recursion limits, depending on the function. The current code already sets a recursion limit above the default, but it has the side effect of silently stomping higher recursion limits set by the user in calling code.